protocol/backend/backend.go

```diff
@@ -337,6 +339,26 @@ func (b *backend) WaitForReceipt(ctx context.Context, txHash ethcommon.Hash) (*e
 	return nil, errReceiptTimeOut
 }
 
+func (b *backend) WaitForTimestamp(ctx context.Context, ts time.Time) error {
+	hdr, err := WaitForEthBlockAfterTimestamp(ctx, b.ethClient, ts.Unix())
+	if err == nil {
+		log.Debug("Wait complete for block %d with ts=%s >= %s",
+			hdr.Number.Uint64(),
+			time.Unix(int64(hdr.Time), 0).Format(common.TimeFmtSecs),
+			ts.Format(common.TimeFmtSecs),
+		)
+	}
+	return err
```

Comment:
```suggestion
	hdr, err := WaitForEthBlockAfterTimestamp(ctx, b.ethClient, ts.Unix())
	if err != nil {
	    return err
	}
	
        log.Debug("Wait complete for block %d with ts=%s >= %s",
              hdr.Number.Uint64(),
              time.Unix(int64(hdr.Time), 0).Format(common.TimeFmtSecs),
              ts.Format(common.TimeFmtSecs),
        )
	return nil
```

is there an `err != nil` case?